lib/Dialect/FIRRTL/Transforms/InjectDUTHierarchy.cpp

```diff
@@ -165,10 +171,22 @@ void InjectDUTHierarchy::runOnOperation() {
     wrapper = dut;
     dut = newDUT;
 
-    // Finish setting up the wrapper.  It can have no annotations.
+    // Finish setting up the wrapper.  Strip the `MarkDUTAnnotation` if we are
+    // in "moveDut" mode.
     AnnotationSet::removePortAnnotations(wrapper,
                                          [](auto, auto) { return true; });
-    AnnotationSet::removeAnnotations(wrapper, [](auto) { return true; });
+    AnnotationSet::removeAnnotations(wrapper, [&](Annotation anno) {
+      if (anno.isClass(dutAnnoClass))
+        return !moveDut;
+      return true;
+    });
+
+    // Finish setting up the DUT.  Strip the `MarkDUTAnnotation` is we are in
+    // "moveDut" mode.
+    if (moveDut)
+      AnnotationSet::removeAnnotations(dut, [&](Annotation anno) {
+        return anno.isClass(dutAnnoClass) && moveDut;
+      });
```

Comment:
nit: I think moveDut is always true in lambda, so simply
```suggestion
    if (moveDut)
      AnnotationSet::removeAnnotations(dut, dutAnnoClass);
```